File: leetcode/easy/Q234_PalindromeLinkedList.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        temp = head
        nodes = []
        while temp:
            nodes.append(temp)
            temp = temp.next

        length = len(nodes)
        for i in range(length):
            if nodes[i].val != nodes[length-1-i].val:
                return False

        return True

File: src/easy/Q234_PalindromeLinkedList.java
package easy;

import java.util.ArrayList;
import java.util.List;

public class Q234_PalindromeLinkedList {
    class ListNode {
        int val;
        ListNode next;
        ListNode(){}
        ListNode(int x) {
            val = x;
            next = null;
        }
    }
    class Solution {
        public boolean isPalindrome(ListNode head) {
            ListNode node = head;
            int length = 0;
            List<Integer> arr = new ArrayList<>();
            while (node != null){
                arr.add(node.val);
                node = node.next;
                length += 1;
            }
            int mid = length / 2;
            for (int i = 0; i < mid; i++) {
                if(!arr.get(i).equals(arr.get(length-i-1))) return false;
            }
            return true;
        }
    }
}
